### DISCUSSION TITLE
Valore negativi immissione in rete da FV

### DISCUSSION THREAD

**Discussion by FlavioRadice** [indomus/forum#2797] [GENERALE - Discussioni generiche]
Ciao, non riesco proprio a venirne fuori, le ho provate tutte dai forum....
Vorrei mettere a zero il valoro di "immissione in rete" nel momento in cui è negativo

...
template:
    - sensor:
        name: 'immissione_in_rete'
        device_class: energy
        state_class: total_increasing
        unit_of_measurement: 'KWh'
        state: >-
            {{ (states('sensor.energymanager_channel_1_power')|float +
            states('sensor.energymanager_channel_2_power')|float )*-1 |round(2) }}
...

**Comment by Borghese** [comment]
ciao,
io sono nuovo, però ho da poco terminato di configurare il monitoraggio di produzione e consumo tramite uno shelly em, e non mi è chiarissima la tua situazione.
Nel senso, la fase di prelievi, durante il periodo di non produzione del FV, non può essere negativa, al max stai a zero, piu facile sia positiva e quindi tu stia prelevando...
Che sistema usi per monitorare produzione e scambi ?

**Reply by FlavioRadice** [reply]
Ciao, uso anch'io SHELLY, ma ho solo due pinze, quindi misuro la produzione e il consumo e l'immissione in rete lo calcolo come differenze, per questo motivo mi serve un aiuto per mettere a zero il valore se negativo

**Reply by Borghese** [reply]
ok allora sei nella mia stessa condizione.
Dovresti usare una pinza da posizionare dopo il contatore di produzione, con direzione K-->L verso il contatore di scambio
La seconda pinza la metti in uscita dal contatore di scambio con direzione K-->L verso le utenze per intenderci

In questo modo, la seconda pinza che posizioni sul contatore di scambio, misurerà in automatico se sono consumi (valori positivi) o immissioni (valori negativi)
non è necessario fare alcuna sottrazione

Otterrai un risultato del genere per intenderci :

Real Time : 
<a href="https://imgbb.com/"><img src="https://i.ibb.co/64PvCS0/1.png" alt="1" border="0"></a>

Storico:
<a href="https://imgbb.com/"><img src="https://i.ibb.co/GHMjwTs/2.png" alt="2" border="0"></a>

**Reply by FlavioRadice** [reply]
Grazie, ma a me occorre solo qualcuno che mi spieghi cortesemente come mettere a zero un valore negativo, non cambio nulla

**Reply by KC-inDomus** [reply]
https://indomus.it/progetti/integrare-e-contabilizzare-un-pannello-solare-fotovoltaico-plug-and-play-su-home-assistant/#riconsegna

**Reply by FlavioRadice** [reply]
Grazie, ma ti chiedo semplicemente se per favore, puoi dirmi come impostare a zero il sensore 'immissione_in_rete" qualora negativo in questa sequenza di istruzioni 

...
template:
- sensor:
name: 'immissione_in_rete'
device_class: energy
state_class: total_increasing
unit_of_measurement: 'KWh'
state: >-
{{ (states('sensor.energymanager_channel_1_power')|float +
states('sensor.energymanager_channel_2_power')|float )*-1 |round(2) }}
...

**Reply by Scialla** [reply]
se leggessi quanto linkato, c'è scritto
